File: JewelsAndStones.cpp
class Solution {
public:
    int numJewelsInStones(string J, string S) 
    {
        set<char> s1;
        int count = 0;
        for(int i = 0; i < J.length(); i++)
        {
            s1.insert(J[i]);
        }
        
        for(int i = 0; i < S.length(); i++)
        {
            if(s1.find(S[i]) != s1.end())
            {
                count = count + 1;
            }
        }
        
        return count;
    }
};

File: JewelsAndStones.py
class Solution(object):
    def numJewelsInStones(self, J, S):
        """
        :type J: str
        :type S: str
        :rtype: int
        """
        
        count = 0
        
        dic = {}
        
        for i in J:
            
            dic[i] = 1
        
        for i in S:
            
            if i in dic:
                
                count = count + 1
                
        return count